code/modules/mob/living/living.dm

```diff
@@ -17,6 +17,7 @@
 		meat_amount = size
 
 	immune_system = new (src)
+	addicted_chems = new /datum/reagents(1000)
```

Comment:
Making this a reagents datum seem like it has too much overhead.  You may want to consider a simple list of reagent types with a mapping to severity.

Reply:
It's needed for on_withdrawal to function

Reply:
It doesn't have to be initialized in `New()` for all `/mob/living`, though. Not even `reagents` is. There's no reason for mobs that can't even contain reagents to carry around this datum.
Just put this in `/mob/living/create_reagents()`.

Reply:
Noted, thanks

Reply:
Done